### PR TITLE
Double-click tab to set title

### LINKED ISSUE #436: [Feature Request] double-click a tab to change its title
I want to suggest a small but impactful (to me, at least) quality-of-life improvement: double-clicking a tab to change it's title.
`xfce4-terminal` has this, and I have found it to be of great value, since it saves time compared to right-clicking and selecting 'Set Title'.

#### Expected behaviour
Double-clicking on a tab should switch to that tab, then invoke the "Set Title" dialog.

### PR DESCRIPTION
**Pull request by Gydo194**
Allow the user to change the title of a tab by double-clicking it.
This closes #436.

### CODE REVIEW

**Review by lukefromdc — APPROVED** [review]
This is simple and it works, approving

**Comment by zhuyaliang** [comment]
squash  commit

**Comment by lukefromdc** [comment]
The "squash and merge" button here should be able to handle that, unless it would screw up the commit message

**Comment by raveit65** [comment]
It's possible edit commit message when using `squash and merge button.
Don't forget to remove the link from commit header which would be added by this procedure :)

File: src/terminal-window.c [hunk 1]
```diff
@@ -2991,6 +2991,15 @@ notebook_button_press_cb (GtkWidget *widget,
         }
     }
 
+    /* If the event is a double click, display the set title dialog */
+    if (event->type == GDK_DOUBLE_BUTTON_PRESS)
+    {
+      terminal_set_title_callback (NULL, window);
```
Comment (zhuyaliang) [inline src/terminal-window.c:2997]: Hold the same indentation

Comment (Gydo194) [reply src/terminal-window.c]: Good catch, fixed it!

**Review by zhuyaliang — APPROVED** [review]
Revised code format, agreed

**Review by raveit65 — APPROVED** [review]
LGTM

**Comment by junfff** [comment]
how can I By configuration file config file to disable this .

**Comment by lukefromdc** [comment]
MATE uses gsettings not configuration files, and there is no setting to turn this on and off at this point. The main reported bug associated with it has been squashed. If someone wants to make this selectable we will review and consider it for merge though

**Pull request merged**